MIQ-Booking-Assistance/index.js

```diff
@@ -107,11 +112,28 @@ async function prepareAndCheckPage(page) {
             snd.loop = true;
             snd.play();
         }
-    }, month);
+    }, thisMonth);
 
     if (found) {
-        console.log("AVAILABLE! Found at: " + new Date().toLocaleString())
-    } else {
+        // functionality to loop through dates in the list and click the first one that is available
+            for (let x in myDates) {
```

Comment:
Could this logic be moved to its own function? This function is already getting a bit out of hand.

Reply:
I'm sure it could be - sorry I'm really not that familiar with JS concepts.  Where would you normally locate functions?  Should it be outside the async function (prepare and check page)?

Reply:
No worries, maybe we should merge it in the current state and then I can help rearrange it later.

Reply:
I would like to merge this soon.

But after merging @alexDrinkwater s Electron App branch there are merge conflicts. Also, I am wondering if we are able to keep the ability to search across a whole month or even better allow users to enter a time frame (from date1 to date2).

I don't mind if the code is not beautiful in the first place, but I think if we just merge it now it's probably going to break things slightly. Unfortunately, I don't have enough time to work on this myself.

@alexDrinkwater would you be able to help to improve this and prepare a merge? That would be fantastic.

Thanks both of you!

Reply:
This is a related request for multiple months: https://github.com/jvolker/miq-nz-booking-tools/issues/5

Reply:
@alexDrinkwater I think I have found another approach to finding the dates. We can use the same attribute that the chrome extension uses and then split the string with an _ this gives us all the dates available. I am working on a PR for that now.